android/sdl_android/src/main/java/com/smartdevicelink/encoder/VirtualDisplayEncoder.java

```diff
@@ -203,9 +205,15 @@ public void run() {
                     inputSurface = prepareVideoEncoder();
 
                     // Create a virtual display that will output to our encoder.
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && virtualDisplay != null) {
+                    virtualDisplay.setSurface(inputSurface);
+                }
+                else {
+                    // recreate after stop in most of cases
                     virtualDisplay = mDisplayManager.createVirtualDisplay(TAG,
-                            streamingParams.getResolution().getResolutionWidth(), streamingParams.getResolution().getResolutionHeight(),
+                            predefinedWidth, predefinedHeight,
```

Comment:
Why was this changed to use 1920x1080 instead of the resolution defined in the streaming parameters?

Reply:
@RHenigan there are platform limitations on changing the size of the VirtualDispay itself. To make the display resizable we should avoid recreating it. Hence there was decided to create the display with 1920x1080 parameters and resize the view inside. I believe the 1920x1080 canvas resolution should cover all possible HMI displays

Reply:
@kostyaBoss This is untrue. There could, for example, be portrait orientation streams: https://cdn.arstechnica.net/wp-content/uploads/2021/02/2021-Ford-Mustang-Mach-E-charging-screens-1-1440x1920.jpg

Reply:
@joeljfischer Thanks for sharing this, let me explain the case which is the root cause. While testing it was discovered that there might come from the HMI resolution bigger than actual (which was used to start the streaming process). If this happens, according to Android platform specification and proposal's needs (do not stop the streaming process and restart it/do not stop encoding) we should not set the new size to VirtualDisplay object, but only work with the view inside. If the view is smaller comparing to VirtualDisplay canvas on the encoding level the "black" pixels will be filtered out by codec. Hence, it was decided to hardcode the 1920x1080 resolution for starting the display. 

I can rollback this to previous values (getting initial w/h from capability) but in this case, if the new capabilities will come with a higher resolution comparing to VD w/h defined previously the image will be cropped by the width of the VD

Previously, while testing this implementation internally there was an input that on HMI there will be no control on the user input and the user can set the new capability with higher resolution eventually

Should I revert the change? Please, let me know the next steps decided regarding this item

Reply:
Hello @kostyaBoss per the proposal the video streaming process should be stopped and restarted. and this change should be reverted

https://github.com/smartdevicelink/sdl_evolution/blob/master/assets/proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle/Resolution_Switching.svg